src/lib/swapper/tradeExecution.ts

```diff
@@ -0,0 +1,95 @@
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+import EventEmitter from 'events'
+import { TRADE_POLL_INTERVAL_MILLISECONDS } from 'components/MultiHopTrade/hooks/constants'
+import { withFromOrXpub } from 'components/MultiHopTrade/hooks/useTradeExecution/helpers'
+import { poll } from 'lib/poll/poll'
+
+import { swappers } from './constants'
+import type { TradeExecutionInput } from './types'
+
+export class TradeExecution extends EventEmitter {
```

Comment:
would be nice to have typed events/listeners for client consumption as a stretch, and I think returning an object might be easier to make sense that the cancel function is being returned (don't get much from the return type to go off).

Reply:
awesome suggestions, will do